### PR TITLE
[1.19.2] Call baked model's `getModelData` before `getRenderTypes`

### PR DESCRIPTION
**Pull request by amadornes**
Currently, the chunk mesher does not call the model's `getModelData(...)` method before calling `getRenderTypes(...)`, and as such the render type query does not have access to any data that may be injected/modified there. However, this data *is* available when calling `getQuads(...)`. There is no reason not to have that modified model data available in both contexts, so this PR makes some trivial changes to the meshing code to reflect that.

Additionally, it ensures that `getModelData(...)` only gets called once per block, not once per block + requested render type, which might lead to a performance uplift in cases where multiple render types and custom model data mutations are used.

### CODE REVIEW

**Review by PepperCode1 — APPROVED** [review]

**Review by marchermans — APPROVED** [review]

**Comment by marchermans** [comment]
@MinecraftForge/triage Can somebody else also look quickly over this?

File: patches/minecraft/net/minecraft/client/renderer/block/BlockRenderDispatcher.java.patch [hunk 1]
```diff
@@ -31,11 +31,14 @@
 +      renderBatched(p_234356_, p_234357_, p_234358_, p_234359_, p_234360_, p_234361_, p_234362_, net.minecraftforge.client.model.data.ModelData.EMPTY, null);
 +   }
 +   public void renderBatched(BlockState p_234356_, BlockPos p_234357_, BlockAndTintGetter p_234358_, PoseStack p_234359_, VertexConsumer p_234360_, boolean p_234361_, RandomSource p_234362_, net.minecraftforge.client.model.data.ModelData modelData, net.minecraft.client.renderer.RenderType renderType) {
```
Comment (Matyrobbrt) [inline patches/minecraft/net/minecraft/client/renderer/block/BlockRenderDispatcher.java.patch:33]: Please depreciate this method

File: patches/minecraft/net/minecraft/client/renderer/block/ModelBlockRenderer.java.patch [hunk 2]
```diff
@@ -12,10 +12,13 @@
 +      tesselateBlock(p_234380_, p_234381_, p_234382_, p_234383_, p_234384_, p_234385_, p_234386_, p_234387_, p_234388_, p_234389_, net.minecraftforge.client.model.data.ModelData.EMPTY, null);
 +   }
 +   public void tesselateBlock(BlockAndTintGetter p_111048_, BakedModel p_111049_, BlockState p_111050_, BlockPos p_111051_, PoseStack p_111052_, VertexConsumer p_111053_, boolean p_111054_, RandomSource p_111055_, long p_111056_, int p_111057_, net.minecraftforge.client.model.data.ModelData modelData, net.minecraft.client.renderer.RenderType renderType) {
++      tesselateBlock(p_111048_, p_111049_, p_111050_, p_111051_, p_111052_, p_111053_, p_111054_, p_111055_, p_111056_, p_111057_, modelData, renderType, true);
```
Comment (Matyrobbrt) [inline patches/minecraft/net/minecraft/client/renderer/block/ModelBlockRenderer.java.patch:15]: Please depreciate this method

**Review by Matyrobbrt — APPROVED** [review]

File: patches/minecraft/net/minecraft/client/renderer/block/BlockRenderDispatcher.java.patch (hunk 1 above)
Comment (marchermans) [reply patches/minecraft/net/minecraft/client/renderer/block/BlockRenderDispatcher.java.patch:33]: Why would that need to be deprecated?

File: patches/minecraft/net/minecraft/client/renderer/block/ModelBlockRenderer.java.patch (hunk 2 above)
Comment (marchermans) [reply patches/minecraft/net/minecraft/client/renderer/block/ModelBlockRenderer.java.patch:15]: Why would that need to be deprecated?

**Pull request merged**